test/app-tap/CMakeLists.txt

```diff
@@ -6,3 +6,96 @@ target_link_libraries(module_api msgpuck)
 set(CMAKE_C_FLAGS "-Wall -Wextra -std=c99")
 set(CMAKE_C_FLAGS_DEBUG "-g -O0 -Werror")
 set(CMAKE_C_FLAGS_RELWITHDEBINFO "-O2")
+
+tarantool_make_lua_path(LUA_PATH
+  PATHS
+    ${CMAKE_CURRENT_SOURCE_DIR}/?.lua
+    ${CMAKE_CURRENT_SOURCE_DIR}/lua/?.lua
+    # Needed for test-run/test_run.lua.
+    ${PROJECT_SOURCE_DIR}/test-run/?.lua
+)
+
+tarantool_make_lua_path(LUA_CPATH
+  PATHS
+  ${CMAKE_CURRENT_BINARY_DIR}/?${CMAKE_SHARED_LIBRARY_SUFFIX}
+)
+
+set(TEST_SUITE_NAME "app-tap")
+
+message(STATUS "Add test suite ${TEST_SUITE_NAME}")
+
+# XXX: The call produces both test and target <app-tap-deps>
+# as a side effect.
+_add_test_suite_target(${TEST_SUITE_NAME}
+  LABELS ${TEST_SUITE_NAME}
+  DEPENDS module_api
+)
+
+list(APPEND TEST_ENV
+  "LUA_PATH=${LUA_PATH}"
+  "LUA_CPATH=${LUA_CPATH}"
+)
+
+file(GLOB tests ${CMAKE_CURRENT_SOURCE_DIR} *.test.lua)
+foreach(test_path ${tests})
+  get_filename_component(test_name ${test_path} NAME)
+  # FIXME: By default, GLOB lists directories.
+  # Directories are omitted in the result if LIST_DIRECTORIES
+  # is set to false. New in version CMake 3.3.
+  if(${test_name} STREQUAL ${TEST_SUITE_NAME})
+    continue()
+  endif()
+  set(test_title "test/${TEST_SUITE_NAME}/${test_name}")
+  set(work_dir "${CMAKE_CURRENT_BINARY_DIR}/${test_name}-workdir")
+  file(MAKE_DIRECTORY ${work_dir})
+  set(TEST_CMD "${RM} -f ${work_dir}/* && ${TARANTOOL_BIN} ${test_path}")
+  add_test(NAME ${test_title}
+           COMMAND ${BASH} -c "${TEST_CMD}"
+           WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+  )
+  set(CTEST_FAIL_REGEXP_EFFECTIVE "")
+  # "Error regular expression found in output. Regex=[bad plan]."
+  if(NOT ${test_title} STREQUAL "test/app-tap/tap.test.lua")
+    set(CTEST_FAIL_REGEXP_EFFECTIVE ${CTEST_FAIL_REGEXP})
+  endif()
```

Comment:
I missed this, thanks!

> A lack of the not ok string doesn't mean that these tests work as expected.

Agree. I would move these tests to directories `test/app` and `test/box` because tests are not TAP13 compliant. Any objections?

Reply:
This issue is addressed in PR https://github.com/tarantool/tarantool/pull/10725
Also patch for ctest support for TAP tests removed, I'll create a separate PR.